ui-performance/index.md

```diff
@@ -0,0 +1,484 @@
+so that the app menu allows you to dynamically toggle the overlay,
+as well as enable checks for calls to `saveLayer` and the presence of
+cached images.
+
+<aside class="alert alert-info" markdown="1">
+**Note:** You can’t enable the performance overlay in the Flutter
+Gallery app downloaded from the App Store. That version of the app
+is compiled to release mode (not profile mode), and doesn’t provide
+a menu for enabling or disabling the overlay.
+</aside>
+
+### Identifying problems in the engine graph
+
+If the performance overlay shows red in the engine graph, start by profiling
+the engine (Dart VM), even if the rasterizer graph also shows red.
+To do this, use
+[Observatory](https://dart-lang.github.io/observatory/), Dart’s profiling tool.
+
+#### Displaying Observatory
+
+Observatory provides a statement-level single-stepping debugger for Dart and
+Flutter applications.  It also supports profiling, examining the heap,
+and reporting on code coverage. Observatory's _timeline_ view allows you
+to capture a snapshot of the stack at a moment in time.
+When you open Observatorty's timeline from the Flutter Inspector,
+you'll be using a version that has been customized for Flutter apps.
+
+{% comment %}
+I've talked to tooling SWEs who feel we're not ready to document this.
+{% endcomment %}
+
+Go to Flutter's timeline view in a browser as follows:
+
+<ol markdown="1">
+<li markdown="1">
+Open Observatory using the line chart icon (<img src="images/observatory-timeline-icon.png" alt="zig-zag line chart icon">)
+from the Flutter Inspector.
+
+(Instead, you could open Observatory using the stopwatch icon (<img src="images/observatory-icon.png">),
+but the "view <u>inspector</u>" link takes you to the standard version of the
+timeline, not the version customized for Flutter.)
+
+![IntelliJ Flutter Inspector Window](/images/intellij/visual-debugging.png)
+</li>
+
+<li markdown="1">
+In VS Code, bring up the command palette and enter "observatory".
+Select **Flutter: Open Observatory Timeline** from the list that pops up.
+If this command isn’t available, make sure that the app is running.
+{% comment %}
+[PENDING: Can you go to the timeline directly? If so, which command?]
+{% endcomment %}
+</li>
+</ol>
+
+#### Using Observatory's timeline
+
+<aside class="alert alert-info" markdown="1">
+**Note:** The Observatory UI and Flutter's custom timeline page are currently
+evolving.  For this reason, we aren't fully documenting the UI at this time.
+If you are comfortable experimenting with Observatory, and would like to give
+us feedback, please file [issues or feature
+requests](https://github.com/dart-lang/sdk/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-observatory)
+as you find them.
+</aside>
+
+### Identifying problems in the rasterizer graph
+
+Sometimes a scene results in a layer tree that is easy to construct,
+but expensive to render on the GPU thread. When this happens,
+the engine graph has no red, but the rasterizer graph shows red.
```

Comment:
a meta comment - are engine and rasterizer the best terms here to use for the user? Perhaps `UI thread` and `GPU thread` (from the list of the 4 threads at the start of the doc)

Reply:
I totally agree, @devoncarew.